src/nvim/eval.c

```diff
@@ -10700,7 +10700,7 @@ int modify_fname(char_u *src, bool tilde_file, size_t *usedlen, char_u **fnamep,
     pbuf = NULL;
     // Need full path first (use expand_env() to remove a "~/")
     if (!has_fullname && !has_homerelative) {
-      if ((c == '.' || c == '~') && **fnamep == '~') {
```

Comment:
Vim also has this and this is quite obvious from the code above.

Reply:
Yeah, true. I can send a vim patch.